clade/proto/sync.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package clade.sync;
+
+import "schema.proto";
+
+message DataSyncCommand {
+  // Path to the Delta Table, relative to the root URL of the store below.
+  // Will be created based on the schema of the RecordBatch if doesn't exist.
+  // e.g. path/to/table
+  string path = 1;
+
+  // Root URL (e.g. s3://my-bucket) and connection options for the table's path
+  schema.StorageLocation store = 2;
+
+  // Primary key columns. Must not be empty: the table must always
+  // have a replica identity / PK.
+  repeated string pk_column = 3;
+
+  // Opaque string identifying data source
+  string origin = 4;
+
+  // Monotonically-increasing transaction number (e.g. the LSN)
+  uint64 sequence_number = 5;
+
+  // True if this is the last command in the transaction denoted
+  // by this LSN and Seafowl can advance its LSN after flushing
+  // this command.
+  bool last = 6;
+}
+
+message DataSyncResult {
+  // If false, Seafowl is overloaded and can't accept the change (e.g.
+  // can't flush fast enough). The client should wait and retry.
+  // TODO: also schema mismatch errors
+  bool accepted = 1;
+
+  // Sequence number up to which the changes are in Seafowl's memory.
+  optional uint64 memory_sequence_number = 2;
```

Comment:
Why do you return this?

Reply:
If you're referring to the fact that it is optional, this is so because after a restart we have nothing in memory, and it feels inadequate to convey this through some actual number (e.g. 0 or -1).

Otherwise if you mean why return this at all, I think it's useful for when e.g. the client (sending) side restarts and wants to replay something—it can then skip some stuff as per this value.

Reply:
Will also be useful later on as per the discussion in 

https://docs.google.com/document/d/1v1HCaw-__fx8JM5DwRe23lc4pt_w8XAC-wRkFoegrbc/edit?disco=AAABKFZw9wA

> the system in general needs to know what's in memory to be able to provide reasonable snapshots for queries